torchvision/csrc/io/image/cpu/decode_png.cpp

```diff
@@ -52,14 +54,20 @@ torch::Tensor decode_png(
 
   struct Reader {
     png_const_bytep ptr;
+    png_size_t count;
   } reader;
   reader.ptr = png_const_bytep(datap) + 8;
+  reader.count = datap_len - 8;
 
   auto read_callback =
       [](png_structp png_ptr, png_bytep output, png_size_t bytes) {
         auto reader = static_cast<Reader*>(png_get_io_ptr(png_ptr));
+        if (reader->count < bytes) {
+          TORCH_CHECK(false, "Out of bound read in decode_png");
+        }
```

Comment:
```suggestion
        TORCH_CHECK(reader->count == bytes, "Out of bound read in decode_png. Probably, the input image is corrupted");
```
@sigseg5 can we do use check directly to check the counter and improve the message by hinting what could be the issue ?